### ISSUE TITLE
iOS crashes sometimes if page is removed while position change has not finished

### ISSUE THREAD

**Issue by xmmk** [alexrainman/CarouselView#116]
Maybe the Element is null in SetViewControllers completion block:

System.NullReferenceException: Object reference not set to an instance of an object
  at CarouselView.FormsPlugin.iOS.CarouselViewRenderer+<SetCurrentPage>c__AnonStorey4.<>m__0 (System.Boolean s) [0x00016] in <8c2f4777355e4928805b606de91d118c>:0 
  at ObjCRuntime.Trampolines+SDUICompletionHandler.Invoke (System.IntPtr block, System.Boolean finished)

**Comment by alexrainman** [comment]
How i can reproduce this?

**Comment by xmmk** [comment]
I can't provide some example code at the moment, so here is a brief description:

I have a ListView inside a ContentPage, which is pushed into a NavigationPage, which is the detail of a MasterDetail page. The ListView header is a StackLayout with the CarouselViewControl as a Child. 

I set up a timer which increases (or resets to 0) the Position every 5 seconds (for crash reproduction 0.5 sec. are better) on the main thread. If I set a new page as MasterDetail detail page while the carousel has just finished (or is about to) the slide animation, the crash occurs quite often.
